### PR TITLE
Fix ug 2

### PR DESCRIPTION
**Pull request by rajobasu**

### CODE REVIEW

File: src/main/java/seedu/address/logic/filters/AddressFilter.java [hunk 1]
```diff
@@ -1,14 +1,34 @@
...
-public class AddressFilter extends AbstractFilter {
+/**
+ * This class is used to filter for the {@code Address} field of the {@code Customer}, based on checking whether the
+ * given filter string is contained inside the address.
+ */
+public class AddressFilter extends Filter {
+
+    /**
+     * Creates a address filter based on a filter string
+     * @param filterString the given filter string against which to match customers
```
Comment (OhJunMing) [inline src/main/java/seedu/address/logic/filters/AddressFilter.java:15]: Spacing between statement and param

File: src/main/java/seedu/address/logic/filters/combinator/FilterCombinator.java [hunk 2]
```diff
@@ -136,6 +155,7 @@ private Node createTree(String description) {
         return formTreeFromNodeStack(nodeStack);
     }
 
+    // TODO
```
Comment (nighoggDatatype) [inline src/main/java/seedu/address/logic/filters/combinator/FilterCombinator.java:158]: More mysterious TODO marker

File: src/main/java/seedu/address/logic/filters/Filters.java [hunk 3]
```diff
@@ -66,22 +77,16 @@ public static AbstractFilter getCorrespondingFilter(String info) {
...
+        return new Filter(info) {
             @Override
             public boolean test(Customer customer) {
                 return false;
             }
-
-            @Override
-            public List<Customer> filterAllCustomers(List<Customer> customer) {
-                return null;
-            }
```
[the remark this answers is not in the document]
Comment (nighoggDatatype) [reply src/main/java/seedu/address/logic/filters/Filters.java:84]: Also, your TODO comment is unexplained. And if you are confused why I want you to throw an exception, this is to follow best practice as taught in CS2103T, to always throw an exception within the 'default' clause.

File: src/main/java/seedu/address/logic/filters/CarFilter.java [hunk 4]
```diff
@@ -21,17 +21,12 @@ public CarFilter(String filterString) {
...
      */
     public static Car parseCar(String car) {
-        String trimmedCar = car;
-        String[] carDetails = trimmedCar.split("\\+");
+        String[] carDetails = car.split("\\+");
 
-
-        if (carDetails.length != 2) {
-            //throw new ParseException(Car.MESSAGE_CONSTRAINTS);
-        }
```
Comment (nighoggDatatype) [inline src/main/java/seedu/address/logic/filters/CarFilter.java:34]: Should this be deleted, or should we throw an exception upon `carDetails.length > 2`

**Review by nighoggDatatype — CHANGES_REQUESTED** [review]
I have concerns with the UserGuide, and I would like you to place String literals in static constants where possible. 

The other stuff can be dealt with in new issues, if we must. Them being:
* The random TODOs
* The bad default case in Filters.java, with an 's'
* The deleted `CarFilter` error handling
* The `Filters` class (which could be renamed to `FilterFactory`)
* The use of `contains` rather than `startsWith` in `TagFilter`

**Review by nighoggDatatype — APPROVED** [review]
LGTM

**Pull request merged**